### PR TITLE
Add access to SSL peer certificate

### PR DESCRIPTION
**Pull request by rix0rrr**
Provide access to a client's certificate information, which makes client
cert based auth/cert SSO a lot more useful.

### CODE REVIEW

File: source/vibe/stream/openssl.d [hunk 1]
```diff
@@ -143,6 +146,25 @@ final class OpenSSLStream : SSLStream {
 		checkExceptions();
 	}
 
+    /** Read certificate info into the clientInformation field */
+    private void readPeerCertInfo(X509 *cert)
+    {
+        X509_NAME* name = X509_get_subject_name(cert);
+
+        int c = X509_NAME_entry_count(name);
+        for (int i = 0; i < c; i++) {
```
Comment (s-ludwig) [inline source/vibe/stream/openssl.d]: Just a little bit shorter and more D-ish would be `foreach (i; 0 .. c)`

File: source/vibe/stream/openssl.d [hunk 2]
```diff
@@ -266,6 +288,11 @@ final class OpenSSLStream : SSLStream {
 			throw m_exceptions[0];
 		}
 	}
+
+    @property SSLCertificateInformation peerCertificate()
+    {
+        return m_peerCertificate;
+    }
```
Comment (s-ludwig) [inline source/vibe/stream/openssl.d]: Can this be `const`? Or does `DictionaryList` complain about the copy then?

**Comment by s-ludwig** [comment]
Looks good to merge, except that lines are indented using spaces instead of tabs. Depending on what editor you use, there might be an [EditorConfig](http://editorconfig.org/) plugin, which is very useful for such formatting issues.

**Comment by rix0rrr** [comment]
Changed spaces to tabs

**Comment by s-ludwig** [comment]
Thanks! Merging.

**Comment by SerialVelocity** [comment]
@s-ludwig @rix0rrr Is there a D client library which allows setting the client certificate? (WebSocket 2-way SSL is actually what I'm looking for)

**Pull request merged**